### PR TITLE
Fix/eslint errors

### LINKED ISSUE #375: fix eslint errors
there are a lot of eslint errors, we should get that under control before the codebase gets any bigger

### PR DESCRIPTION
**Pull request by shanejonas**
image
image

no longer does our travis ci require `allow_errors` when linting 🎉 

still needs some work with 91 warnings. but this is a good start and passes travis ci.

fixes #375

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -16,10 +20,14 @@
     }
   },
   "rules": {
-    "indent": ["error", 4, {
+    "indent": ["error", 2, {
       "SwitchCase": 1
     }],
+    "react/prop-types": 0,
```
Comment (shanejonas) [inline .eslintrc]: I will be adding back prop types in the future

File: .eslintrc [hunk 2]
```diff
@@ -28,8 +36,13 @@
       "exports": "ignore",
       "functions": "never"
     }],
+    "no-use-before-define": 0,
+    "import/no-named-as-default": 0,
     "import/prefer-default-export": 1,
-    "max-len": ["error", {"code": 120}],
+    "max-len": 0,
+    "class-methods-use-this": 0,
```
Comment (shanejonas) [inline .eslintrc]: I'd like to add this back in the future, if we dont use `this`, its probably better off as a stateless react component (as just a function)

**Comment by splix** [comment]
We use problem/solution type comments, so for you commit it would be more helpful to have:

fix(eslint errors) -> 
problem: eslint errors

fix(eslint + indenting) -> 
problem: non standard indent 
solution: use 4 spaces

fix(eslint autofix) ->
solution: eslint autofix

and so on. Otherwise it's meaningless, what `fix(react proptypes)` means? How it will help following developer? 
Or `fix(add ethereumjs-util to package.json)`. Not clear from comment. does it solve anything, or is it required by some code? It doesn't provide anything new to the commit itself, I see that a dependency was introduce, but nothing more, so if everything was working before why it was introduced? If something didn't work before this commit the comment sections is designed exactly for this, to mention the reason of the commit. It's impossible to guess otherwise.
Put yourself on a place of someone who debugs/refactor/cleanup this code written 2 years ago, and you found such commit. Will it be helpful for you?

**Comment by shanejonas** [comment]
I've been mostly following this angular's commit message guideline:

https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits

Usually i would squash this down to 1 commit called `fix(eslint errors)` for a PR but having so many lines of code change it made sense to break it up.

This lets you auto generate changelog's.

Do we have a similar tool for the problem/solution? how do you tell the difference between a bug fix and a feature implementation vs refactoring? do we make the changelogs by hand currently?

**Comment by splix** [comment]
It's from Collective Code Construction Contract introduced in ZeroMQ https://rfc.zeromq.org/spec:42/C4/

See 2.3. Patch Requirements:
> 7. A patch commit message MUST consist of a single short (less than 50 characters) line stating the problem ("Problem: ...") being solved, followed by a blank line and then the proposed solution ("Solution: ...").

Although we don't follow C4 strictly (for example it requires that code must be under GPL), we found this practice of Problem/Solution for commit message very useful.

**Comment by shanejonas** [comment]
Ok, thanks for the spec link. all fixed up.

**Comment by splix** [comment]
Also, from my experience, I omit one part of this pair when it doesn't provide extra information. Like for "_problem: eslint errors / solution: fix errors_" second part is obvious in this context, so sometimes I skip it, keeping only "_problem: eslint errors_" in a message. That's ok too imho.
Your "_solution: eslint autofix + change files to pass linting_" is helpful, though, it gives extra information that it was done automatically. So if you see you can provide meaningful message for both part, please do it.

**Pull request merged**